virttest/qemu_devices/qcontainer.py

```diff
@@ -2037,7 +2037,10 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
                 if Flags.BLOCKDEV in self.caps:
                     if key == 'discard':
                         value = re.sub('on', 'unmap', re.sub('off', 'ignore', value))
-                    devices[-2].set_param(key, value)
+                    if key == 'cache-size':
+                        devices[-2].set_param(key, None)
+                    else:
+                        devices[-2].set_param(key, value)
```

Comment:
How about? 
```
            for key, value in drv_extra_params:
                if Flags.BLOCKDEV in self.caps:
                    if key == 'discard':
                        value = re.sub('on', 'unmap', re.sub('off', 'ignore', value))
                    elif key == 'cache-size':
                        continue
                    devices[-2].set_param(key, value)
                devices[-1].set_param(key, value)
```

Reply:
If like this, it will skip set cache-size to qcow2 node. Just need to skip driver=file (protocol line).

the correct cmd lines:
-blockdev node-name=file_image1,driver=file,aio=threads,filename=/home/kvm_autotest_root/images/rhel830-64-virtio-scsi.qcow2,cache.direct=on,cache.no-flush=off \
-blockdev node-name=drive_image1,driver=qcow2,cache.direct=on,cache.no-flush=off,file=file_image1,cache-size=0 \

Reply:
I misread, thanks for your explanation.